File: internal/cliclient/build_env.go
package cliclient

import (
	"context"
	"net/url"
)

// ListBuildEnvVars returns current-user build environment variable metadata.
func (c *Client) ListBuildEnvVars(ctx context.Context) ([]BuildEnvVar, error) {
	var vars []BuildEnvVar
	_, err := c.Get(ctx, "/build-env-vars", &vars)
	if err != nil {
		return nil, err
	}
	return vars, nil
}

// UpsertBuildEnvVar creates or updates a current-user build environment variable.
func (c *Client) UpsertBuildEnvVar(ctx context.Context, req UpsertBuildEnvVarRequest) (*BuildEnvVar, error) {
	var envVar BuildEnvVar
	_, err := c.Put(ctx, "/build-env-vars", req, &envVar)
	if err != nil {
		return nil, err
	}
	return &envVar, nil
}

// DeleteBuildEnvVar deletes a current-user build environment variable by key.
func (c *Client) DeleteBuildEnvVar(ctx context.Context, key string) error {
	_, err := c.Delete(ctx, "/build-env-vars/"+url.PathEscape(key))
	return err
}

File: frontend/src/api/buildEnv.ts
import type { BuildEnvVar, UpsertBuildEnvVarRequest } from '@/types';
import { apiClient } from './client';

export type BuildEnvTarget = 'local' | 'remote';

const buildEnvBasePath = (target: BuildEnvTarget) =>
  target === 'remote' ? '/remote/build-env-vars' : '/build-env-vars';

export const buildEnvApi = {
  list: async (target: BuildEnvTarget): Promise<BuildEnvVar[]> => {
    const { data } = await apiClient.get(buildEnvBasePath(target));
    return data;
  },

  upsert: async (
    target: BuildEnvTarget,
    req: UpsertBuildEnvVarRequest,
  ): Promise<BuildEnvVar> => {
    const { data } = await apiClient.put(buildEnvBasePath(target), req);
    return data;
  },

  delete: async (target: BuildEnvTarget, key: string): Promise<void> => {
    await apiClient.delete(
      `${buildEnvBasePath(target)}/${encodeURIComponent(key)}`,
    );
  },
};
